pkg/service/generateConfig/generateConfig.go

```diff
@@ -47,17 +46,17 @@ func (g *generatorConfig) GenerateConfig(filePath string, options GenerateConfig
 
 	finalOutput := append(results, []byte(utils.ConfigGuide)...)
 
+	err = utils.SetUmask(0)
```

Comment:
why are you first setting to 0 and then 0022. What is the problem in directly setting to 0022?

Reply:
After creating the file, the umask is restored to 0022 to apply the default permissions for subsequent file creations.